internal/backend/db/dbgorm/ownership.go

```diff
@@ -185,17 +198,25 @@ func (gdb *gormdb) withUserEntity(ctx context.Context, entity string) *gorm.DB {
 }
 
 type OwnershipChecker interface {
-	IsUserEntity(db *gorm.DB, user string, ids ...sdktypes.UUID) error
+	EnsureUserAccessToEntities(db *gorm.DB, user string, ids ...sdktypes.UUID) error
+	EnsureUserAccessToEntitiesWithOwnership(db *gorm.DB, user string, ids ...sdktypes.UUID) ([]scheme.Ownership, error)
 	JoinUserEntity(db *gorm.DB, entity string, user string) *gorm.DB
 }
 
 type UsersOwnershipChecker struct {
 	z *zap.Logger
 }
 
-func (c *UsersOwnershipChecker) IsUserEntity(db *gorm.DB, uid string, ids ...sdktypes.UUID) error {
+func (c *UsersOwnershipChecker) EnsureUserAccessToEntitiesWithOwnership(
+	db *gorm.DB, uid string, ids ...sdktypes.UUID,
+) ([]scheme.Ownership, error) {
```

Comment:
you return here ```[]scheme.Ownership``` but in actual you return only part of the objects, since you select only user_id and event_type, missing the entity_id. This can easily lead to mistakes next time someone uses it. 

wdyt ?

Reply:
fixed. returning full model now